interface/drugs/add_edit_lot.php

```diff
@@ -280,34 +477,50 @@ function trans_type_changed() {
             if ($form_quantity < 0) {
                 $info_msg = xl('Transaction failed, quantity is less than zero');
             } else {
-                $lot_id = sqlInsert(
-                    "INSERT INTO drug_inventory ( " .
-                    "drug_id, lot_number, manufacturer, expiration, " .
-                    "vendor_id, warehouse_id, on_hand " .
-                    ") VALUES ( " .
-                    "?, "                            .
-                    "?, " .
-                    "?, " .
-                    "?, "  .
-                    "?, " .
-                    "?, " .
-                    "? "  .
-                    ")",
-                    array(
-                        $drug_id,
-                        $_POST['form_lot_number'],
-                        $_POST['form_manufacturer'],
-                        (empty($_POST['form_expiration']) ? "NULL" : $_POST['form_expiration']),
-                        $_POST['form_vendor_id'],
-                        $_POST['form_warehouse_id'],
-                        $form_quantity
-                    )
+                $exptest = $form_expiration ?
+                    ("expiration = '" . add_escape_custom($form_expiration) . "'") : "expiration IS NULL";
```

Comment:
generally should avoid mixing binding and manual escaping, however this specific case should be fine (since there is no way that a `?` character will find its way in this variable)

Reply:
OK thanks.